pkg/libursa/ursa/ursa.go

```diff
@@ -119,6 +119,28 @@ func CredentialPublicKeyFromJSON(jsn string) (unsafe.Pointer, error) {
 	return handle, nil
 }
 
+func CredentialPublicKeyToJSON(pubKey unsafe.Pointer) ([]byte, error) {
+	var pubJson *C.char
+	defer C.free(unsafe.Pointer(pubJson))
+	result := C.ursa_cl_credential_public_key_to_json(pubKey, &pubJson)
+	if result.code != 0 {
```

Comment:
You need to move the defer to after the result.code check.  Otherwise you could be free'ing a pointer that hasn't been allocated.